detox/src/client/actions/SyncStatusFormatter.js

```diff
@@ -0,0 +1,186 @@
+    `${resourceSubSubTitle(`Repeat interval: ${timer.repeat_interval}`)}.\n` +
+    `${resourceSubSubTitle(`Is recurring: ${timer.is_recurring ? `YES` : `NO`}`)}.`;
+}
+
+function resourceSubTitle(string) {
+  return `  - ${string}`;
+}
+
+function resourceSubSubTitle(string) {
+  return `    + ${string}`;
+}
+
+const uiResourceCounterNameMapping = {
+  'layer_animation_pending_count':  `Layer animations pending`,
+  'layer_needs_display_count': `Layers needs display`,
+  'layer_needs_layout_count': `Layers needs layout`,
+  'layer_pending_animation_count': `Layers pending animations`,
+  'view_animation_pending_count': `View animations pending`,
+  'view_controller_will_appear_count': `View controllers will appear`,
+  'view_controller_will_disappear_count': `View controllers will disappear`,
+  'view_needs_display_count': `View needs display`,
+  'view_needs_layout_count': `View needs layout`,
+  'reason': `Reason`
+};
+
+function uiResourceDescriptionFromProperties(properties) {
+  let countersDescriptions = [];
+  for (const [key, value] of Object.entries(properties)) {
+    countersDescriptions.push(resourceSubTitle(`${uiResourceCounterNameMapping[key]}: ${value}`));
+  }
+
+  return `${resourceTitle(`UI elements are busy:`)}\n${countersDescriptions.join('.\n')}.`;
+}
+
+function jsTimersResourceDescriptionFromProperties(properties) {
+  let timerCount = 0;
+  let timersDescriptions = [];
+  for (const timer of properties.timers) {
+    timerCount++;
+    timersDescriptions.push(jsTimerDescriptionFromTimer(timer, timerCount));
+  }
+
+  return `${resourceTitle(`${timerCount} enqueued JavaScript timers:`)}` +
+    `\n${timersDescriptions.join('\n')}`;
+}
+
+function jsTimerDescriptionFromTimer(timer, timerCount) {
+  return `${resourceSubTitle(`Timer #${timerCount}:`)}\n` +
+    `${resourceSubSubTitle(`JS timer ID: ${timer.timer_id}`)}.\n` +
+    `${resourceSubSubTitle(`Duration: ${timer.duration}`)}.\n` +
+    `${resourceSubSubTitle(`Is recurring: ${timer.is_recurring ? `YES` : `NO`}`)}.`;
+}
+
+function networkResourceDescriptionFromProperties(properties) {
+  let urlCount = 0;
+  let urlsDescriptions = [];
+  for (const url of properties.urls) {
+    urlCount++;
+    urlsDescriptions.push(urlDescriptionFromURL(url, urlCount));
+  }
+
+  return `${resourceTitle(`${urlCount} network requests with URLs:`)}` +
+    `\n${urlsDescriptions.join('\n')}`;
+}
+
+function urlDescriptionFromURL(url, urlCount) {
+  return resourceSubTitle(`URL #${urlCount}: ${url}.`);
+}
+
+function looperResourceDescriptionFromProperties(properties) {
+  const executionType = properties.execution_type;
+  return resourceTitle(`${properties.thread} is executing` +
+    `${executionType !== undefined ? ` (${executionType}).` : `.`}`);
+}
+
+function unknownResourceDescriptionFromProperties(properties) {
+  return resourceTitle(`Resource "${properties.identifier}" is busy.`);
+}
+
+module.exports = { formatJSONStatus };
```

Comment:
```suggestion
module.exports = formatJSONStatus;
```

(this will have side-effects)

Reply:
can you elaborate please?

Reply:
discussed :+1: